lib/oban/engines/basic.ex

```diff
@@ -324,20 +324,43 @@ defmodule Oban.Engines.Basic do
     end
   end
 
-  defp return_or_replace(conf, query_opts, job, changeset) do
+  defp return_or_replace(conf, query_opts, %{state: state} = job, changeset) do
     case Changeset.get_change(changeset, :replace, []) do
       [] ->
         {:ok, job}
 
-      replace_keys ->
-        Repo.update(
-          conf,
-          Ecto.Changeset.change(job, Map.take(changeset.changes, replace_keys)),
-          query_opts
-        )
+      replace ->
+        if valid_replace_option?(replace) do
```

Comment:
The legacy options are valid too, just deprecated. How about `state_replace_option?`